client/checkout/stripe-link/index.js

```diff
@@ -65,6 +73,11 @@ const enableStripeLinkPaymentMethod = async ( options ) => {
 		);
 		switchToNewPaymentTokenElement();
 	} );
+
+	return () => {
+		emailField.removeEventListener( 'keyup', handleKeyup );
+		removeLinkButton();
+	};
```

Comment:
Discussed offline - we'll need to handle the early return on L54, because it's not returning a promise like the `client/checkout/blocks/payment-processor.js` expects it to when the email field is not present.
You could replace the early return with:
```
return Promise.resolve( () => null );
```

Reply:
Errata corrige: my bad, since `enableStripeLinkPaymentMethod` is defined as `async`, it will always return a promise. All good here! :) sorry about the confusion!

Reply:
You beat me by responding faster! 😄 
Agreed. I think returning a Promise would still be great for consistency https://github.com/Automattic/woocommerce-payments/pull/9822/commits/d87b45266fdbb7dd870390b3962e8579f7725482 👀